tests/phpunit/integration/api/actions/beansReplaceAction.php

```diff
@@ -0,0 +1,271 @@
+			$this->assertFalse( _beans_get_action( $beans_id, 'added' ) );
+
+			// Now store away the "replace" hook.
+			$this->assertFalse( beans_replace_action( $beans_id, null, $replaced_action['callback'] ) );
+
+			// Check that stored.
+			$this->assertEquals( $replaced_action, _beans_get_action( $beans_id, 'modified' ) );
+		}
+	}
+
+	/**
+	 * Test beans_replace_action() should store the "replaced" action when the original action
+	 * has not yet been registered.  Once the original action is registered, then it should be replaced.
+	 *
+	 * Intent: We are testing to ensure Beans is "load order" agnostic.
+	 */
+	public function test_should_store_and_then_replace_action() {
+		$replaced_action = array(
+			'callback' => 'my_new_callback',
+		);
+
+		// Now replace the actions.
+		foreach ( static::$test_ids as $beans_id ) {
+			beans_replace_action( $beans_id, null, $replaced_action['callback'] );
+		}
+
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $original_action ) {
+			// Check if it replaced the callback.
+			$new_action = _beans_get_action( $beans_id, 'added' );
+			$this->assertEquals( $original_action['hook'], $new_action['hook'] );
+			$this->assertEquals( $replaced_action['callback'], $new_action['callback'] );
+			$this->assertEquals( $original_action['priority'], $new_action['priority'] );
+			$this->assertEquals( $original_action['args'], $new_action['args'] );
+
+			// Check that the "replaced" action has been stored in Beans and WordPress.
+			$this->check_stored_in_beans( $beans_id, $replaced_action );
+			$this->check_registered_in_wp( $original_action['hook'], $new_action );
+		}
+	}
+
+	/**
+	 * Test beans_replace_action() should return false when there's nothing to replace,
+	 * i.e. no arguments passed.
+	 */
+	public function test_should_return_false_when_nothing_to_replace() {
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action_config ) {
+			$this->assertFalse( beans_replace_action( $beans_id ) );
+
+			// Verify that it did not get stored in "replaced" or "modified".
+			global $_beans_registered_actions;
+			$this->assertArrayNotHasKey( $beans_id, $_beans_registered_actions['replaced'] );
+			$this->assertArrayNotHasKey( $beans_id, $_beans_registered_actions['modified'] );
+
+			// Check that the original action has not been replaced.
+			$this->assertSame( $action_config, _beans_get_action( $beans_id, 'added' ) );
+		}
+	}
+
+	/**
+	 * Test beans_replace_action() should replace the registered action's hook.
+	 */
+	public function test_should_replace_the_action_hook() {
+		$this->go_to_post();
+
+		foreach ( static::$test_actions as $beans_id => $action_config ) {
+			$original_action = _beans_get_action( $beans_id, 'added' );
+
+			// Make sure the callback is what we think before we get rolling.
+			$this->assertEquals( $action_config['hook'], $original_action['hook'] );
+
+			// Setup what will get stored in Beans.
```

Comment:
`Set up what will get stored in Beans` (rather than `Setup that will get stored in Beans`)